### DISCUSSION TITLE
active_pane.foreground_process_name return empty

### DISCUSSION THREAD

**Discussion by jopcode** [wezterm/wezterm#4772] [Q&A]
### What Operating System(s) are you running on?

Linux X11

### WezTerm version

20240103-143005-4921f139

### Ask your question!

when i run a command with **sudo,** `active_pane.foreground_process_name` his returnis empty

![image](https://github.com/wez/wezterm/assets/16161754/35ead8c4-721b-4252-9e31-9fbc76580dfd)

there is my code for get tab name
```lua
function tab_title(tab_info)
  local title = tab_info.tab_title
  -- if the tab title is explicitly set, take that
  if title and #title > 0 then
    return title
  end
  -- Otherwise, use the title from the active pane
  -- in that tab
  return  string.gsub(tab_info.active_pane.foreground_process_name, "(.*[/\\])(.*)", "%2")
end
```
so, my question is how can I set the tab title when I use sudo for a command in the same way as if I did not use sudo?

**Comment by nekowinston** [comment]
I wouldn't rely on `active_pane.foreground_process_name` here, and use the [Shell Integration](https://wezfurlong.org/wezterm/shell-integration.html?h=WEZTERM_PROG#user-vars) instead.

**Reply by jopcode** [reply]
Could you give me an example please, I can't quite understand the implementation.
